app/api/getProducts/route.ts

```diff
@@ -0,0 +1,51 @@
+// app/api/post/route.ts
+import postgres from '@/app/utils/postgres';
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function POST(request: NextRequest) {
+    try {
+        const client = await postgres.connect();
+
+        const { productName, itemCode, quantity: rawQuantity } = await request.json();
+        const quantity = rawQuantity ? parseInt(rawQuantity) : null;
+
+        if (!productName || !itemCode || !quantity || quantity < 0) {
+            return NextResponse.json({ error: 'Please provide all required fields' }, { status: 400 });
+        }
+        const result = await client.query(
+            `INSERT INTO item (code, name, quantity) VALUES ($1, $2, $3) RETURNING *`,
+            [itemCode, productName, quantity]
+        );
+
+        client.release();
+        return NextResponse.json(result, { status: 200 });
+    }
+    catch (error) {
+        return NextResponse.json({ error }, { status: 500 });
+    }
+}
+
+
+export async function GET(request: NextRequest) {
+    try {
+        const client = await postgres.connect();
+
+        const searchParams = request.nextUrl.searchParams;
```

Comment:
Sorry I didn't catch this in the first review, but I think it would be much more readable to do a query like this instead of using regex to split the item code:

```
SELECT * FROM item WHERE CONCAT(code, id) = '$(realItemCode)'
```

what do you think?

Reply:
A bit unconventional but try it out if it works